symengine/cwrapper.cpp

```diff
@@ -52,6 +52,16 @@ void basic_free(basic s)
     s->m.~RCP();
 }
 
+basic_struct* basic_new_heap()
+{
+    return new CRCPBasic();
```

Comment:
Shouldn't `CRCPBasic()` be changed to `basic_struct`? Since we have used ifdefs to declare two structs `CRCPBasic` and `CRCPBasic_C` based on the compiler. They are ultimately typedeffed to `basic_struct`.